DynamicTablesPkg/Library/Acpi/Common/AcpiSpcrLib/SpcrGenerator.c

```diff
@@ -397,6 +607,34 @@ BuildSpcrTableEx (
     AsciiStrCpyS (AcpiSpcr.NameSpaceString, sizeof (NAME_STR_SPCR_PORT), NAME_STR_SPCR_PORT);
   }
 
+  if (SerialPortInfo->PortSubtype == EFI_ACPI_DBG2_PORT_SUBTYPE_SERIAL_FULL_16550) {

```

Comment:
Last comments:
- would it be possible to move this block after the following block:
```
  // Set the access size
  if (SerialPortInfo->AccessSize >= EFI_ACPI_6_3_QWORD) {
    ...
  }
```
Indeed, it would be better to have all the code modifying `AccessSize` at the same place.
- If the port usbtype is `SERIAL_FULL_16550`, is it possible to have a BYTE accessSize ? Shouldn't we have the legacy value (UNDEFINED) and issue an error if we don't have:
```
(SerialPortInfo->AccessSize == EFI_ACPI_6_3_UNDEFINED)
```